### PR TITLE
Clone JSON RPC request ID

### PR DESCRIPTION
**Pull request by quangkeu95**

### CODE REVIEW

File: node/node_cache.go [hunk 1]
```diff
@@ -148,9 +171,11 @@ func (nc *NodeCache) HandleRequest(req *http.Request) ([]byte, error) {
 	//get message from request body
 	message := JSONRPCMessage{}
 	if err := json.Unmarshal(body, &message); err == nil {
-		if cacheResp := nc.GetCacheResponse(message.Method); cacheResp != nil {
-			return cacheResp, nil
+		cacheResp, respErr := nc.GetCacheResponse(message)
```
Comment (andrewsource147) [inline node/node_cache.go:174]: Should make request if there are error from cache

File: node/node_cache.go [hunk 2]
```diff
@@ -121,20 +137,27 @@ func (nc *NodeCache) makeRequest(method string) (*http.Request, error) {
...
+		// clone user request ID
+		jsonRPCResponse.ID = message.ID
+		result, err := json.Marshal(jsonRPCResponse)
+		if err != nil {
+			return []byte{}, err
+		}
+		return result, nil
 	}
-	return nil
+	return []byte{}, nil
```
Comment (andrewsource147) [inline node/node_cache.go:160]: If we cannot get from cache, we have to return error?

File: node/node_cache.go [hunk 3]
```diff
@@ -48,9 +54,10 @@ func (nc *NodeCache) run() {
 
 // cacheWorker A worker to serve a method
 func (nc *NodeCache) cacheWorker(method string) {
+	id := 1
 	ticker := time.NewTicker(10 * time.Second)
 	for {
-		req, err := nc.makeRequest(method)
+		req, err := nc.makeRequest(method, id)
```
Comment (andrewsource147) [inline node/node_cache.go:60]: Why need to add id here?, we can make request without id. and we dont use this id after all

File: node/node_cache.go (hunk 1 above)
Comment (quangkeu95) [reply node/node_cache.go:171]: ok

File: node/node_cache.go (hunk 2 above)
Comment (quangkeu95) [reply node/node_cache.go]: return error here

File: node/node_cache.go (hunk 3 above)
Comment (quangkeu95) [reply node/node_cache.go]: remove id

**Pull request merged**